### PR TITLE
[38] Changes to add a modal to show a list of players who don't have a mount

### LINKED ISSUE #38: Show users that do not have a mount in the mounts panel
**Is your feature request related to a problem? Please describe.**
- It is hard to track users that do not have a certain mount without going through multiple pages on the user mounts panel.

**Describe the solution you'd like**
- When a user is on the mounts panel and clicks on a mount name, we show a modal to show users that do not have a certain mount.

**Describe alternatives you've considered**
- Be able to filter out users that do not have a certain mount on the user mounts panel, but there are multiple mounts showing on each tab.

**Additional context**
- N/A

### PR DESCRIPTION
**Pull request by sofia819**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bug fix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## Issue
- [38](https://github.com/sofia819/mount_recorder_client/issues/38)

## What is the current behavior?
- Mounts tab does not have interactivity

## What is the new behavior?

- On the mounts tab, when clicking on a mount name, a modal will open and show users who do not have that particular mount.

## Other information
mounts missing

### CODE REVIEW

**Comment by sofia819** [comment]
@EvanM27 If you could suggest better names for the files, components, and variables.

File: src/utils/constants.js [hunk 1]
```diff
@@ -124,6 +124,8 @@ export const ALERT_MESSAGE = {
 
 export const ALERT_MESSAGE_DURATION = 3000;
 
+export const USERS_NOT_OWNED = 'Players who do not have';
```
Comment (EvanM27) [inline src/utils/constants.js]: Better name might be USERS_LACKING or NOT_OWNED_BY_USERS

Comment (EvanM27) [reply src/utils/constants.js]: Alternatively, USERS_WITHOUT, USERS_WITHOUT_MOUNT

File: src/components/Mounts/MountsOwner/MountOwnedUsersList.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import React, { forwardRef } from 'react';
+import { Card, Button, Grid, Typography, Box } from '@material-ui/core';
+import { CLOSE_BUTTON, USERS_NOT_OWNED } from 'utils/constants';
+import PropTypes from 'prop-types';
+import 'components/ModalForm.scss';
+
+export const MountOwnedUsersList = forwardRef((props, ref) => {
```
Comment (EvanM27) [inline src/components/Mounts/MountsOwner/MountOwnedUsersList.js]: Alternatively, MountsUsersLackingList

File: src/components/Mounts/MountsOwner/MountsOwnedButton.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React, { createRef, useState } from 'react';
+import { Button, Modal } from '@material-ui/core';
+import { MountOwnedUsersList } from 'components/Mounts/MountsOwner/MountOwnedUsersList';
+import PropTypes from 'prop-types';
+
+export const MountsOwnedButton = (props) => {
```
Comment (EvanM27) [inline src/components/Mounts/MountsOwner/MountsOwnedButton.js]: Might consider making  this MountsUsersLackingButton

File: src/utils/constants.js (hunk 1 above)
Comment (sofia819) [reply src/utils/constants.js]: Renaming to `NOT_OWNED_BY_USERS`

**Review by EvanM27 — APPROVED** [review]
LGTM

**Pull request merged**